### DISCUSSION TITLE
✅ Handling or omitting recursive schemas from openapi doc generation

### DISCUSSION THREAD

**Discussion by TheWisestOne** [RobinTail/express-zod-api#801] [Ideas]
Hi,

One of the APIs includes a return type that includes a recursive type definition. The way to do this in `zod` is to use the `z.lazy` function and make a self-referential type. 

However, when using express-zod-api to generate `open-api` swagger docs, we get an error

```
Unhandled error. (OpenAPIError: Zod type ZodLazy is unsupported
```

To me it makes sense that generating swagger docs for a recursive type might be hard / obnoxious, but I'm wondering if there's a way for me to still use `express-zod-api` to include this route in my `routing`, but simply omit it from my generated api docs (or better yet, override the output type in my api docs to just be a simplified version that doesn't include the recursive key)?

**Comment by RobinTail** [comment]
Hmmm... that's an interesting problem.
I'm going to think on it a bit...
Sorry for delay, @TheWisestOne

**Reply by TheWisestOne** [reply]
My current approach is just to have my inputSchema use `z.object({}).passthrough()` at the moment the recursion happens (which obviously isn't ideal, but at least lets me continue using the framework in the meantime)

**Comment by RobinTail** [comment]
Perhaps I could use OpenAPI's schema references for describing recursion.

**Reply by TheWisestOne** [reply]
I think in general allowing the use of schema references would be quite useful! There are often a lot of re-used types across an API, and schemas really help consumers understand the relationships between these types and recognize when subtypes are re-used

**Comment by RobinTail** [comment]
I found a way to do it in #856

**Comment by RobinTail** [comment]
🚀  The feature released in v9.3.1, @TheWisestOne
